### PR TITLE
Server console

### PR DESCRIPTION
**Pull request by ajacksified**
:eyeglasses: @curioussavage, @umbrae 

This adds:
* snoode api request logging (via https://git.io/vlxC1)
* active open requests logging stub (so we can track this in statsd later)
* a developer console, enableable with the `--console` flag (`node index.js --console`)

If the console is up, it will write logs to `reddit-mobile.log` in your project directory, clearing it out each time you start up the console.

http://gfycat.com/NegligibleAdorableBat

### CODE REVIEW

File: src/server/console.es6.js [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+    });
+
+    this.emitter.on('logRequest', (data) => {
+      apiLog.log(data);
+    });
+
+    return apiLog;
+  }
+
+  buildMap (grid) {
```
Comment (curioussavage) [inline src/server/console.es6.js]: this is a nice touch here :laughing:

Comment (ajacksified) [inline src/server/console.es6.js]: Gotta have blinky `X`s on a map, or it isn't a real dashboard.

File: src/server/index.jsx [hunk 2]
```diff
@@ -134,6 +136,13 @@ class Server {
         secureProxy: config.httpsProxy
     };
 
+    let that = this;
+    server.use(function * (next) {
+      that.activeRequests++;
+      yield next;
+      that.activeRequests--;
```
Comment (curioussavage) [inline src/server/index.jsx]: Can you tell me how this works?  when does line 143 get called?  (fuzzy understanding of generators here.)

File: package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     "build": "gulp",
     "watch": "gulp watch",
     "dev": "gulp dev",
-    "test": "mocha test/index.js"
+    "test": "mocha test/index.js",
+    "console": "babel-node ./src/server/console.es6.js"
```
Comment (curioussavage) [inline package.json]: when would I use this?  should this be running index.jsx?

**Comment by curioussavage** [comment]
:nail_care:  pretty much a fish, I just have questions, maybe add some inline documentation for the next guy?

File: package.json (hunk 3 above)
Comment (ajacksified) [inline package.json]: Ah, yeah, leftovers.

File: src/server/index.jsx (hunk 2 above)
Comment (ajacksified) [inline src/server/index.jsx]: 143 gets called after all the other generators have finished yielding (basically, at the end of a response, just before it's sent to a browser.)

**Comment by ajacksified** [comment]
:haircut:

**Comment by curioussavage** [comment]
:fish:

**Pull request merged**